python/lsst/obs/base/butler_tests.py

```diff
@@ -186,12 +204,16 @@ def test_flat(self):
     def test_raw_header_wcs(self):
         """Test that `raw_header_wcs` returns the unmodified header of the raw
         image."""
-        if self.butler_get_data.raw_header_wcs is not None:
-            wcs = self.butler.get("raw_header_wcs", self.dataIds["raw"])
-            self.assertEqual(wcs, self.butler_get_data.raw_header_wcs)
+        if self.butler_get_data.raw_header_wcs is None:
+            self.skipTest("Skipping raw header WCS test since no reference provided.")
+        # Gen3 will not understand this at the moment.
+        wcs = self.butler.get("raw_header_wcs", self.dataIds["raw"])
+        self.assertEqual(wcs, self.butler_get_data.raw_header_wcs)
 
-    @unittest.skip("Cannot test this, as there is a bug in the butler! DM-8097")
+    @unittest.skip("Cannot test this. Raw data can not reliably have bbox requested.")
     def test_raw_sub_bbox(self):
+        # Gen2 variant only. Gen3 will never support this.
+        self._require_gen2()
```

Comment:
It's effectively impossible when all the amps are stored separately. It could be done if we had a lot of clever assembly code that worked out which amps it needed to read and then tried to form the subset (or even if we made it read all the amps and then delete it -- it's up to the formatter rather than the butler so a generic test is meaningless. Some raw formatters might support it. I can remove the test completely.